mtbl.pyx

```diff
@@ -31,12 +46,15 @@ def varint_length(uint64_t value):
     """varint_length(v) -> number of bytes the integer v would require in varint encoding."""
     return mtbl_varint_length(value)
 
-def varint_length_packed(bytes py_buf):
+def varint_length_packed(py_buf):
     """varint_length_packed(b) -> number of varint-packed bytes at the start of b."""
     cdef uint8_t *buf
     cdef Py_ssize_t len_buf
     cdef size_t sz
-    PyString_AsStringAndSize(py_buf, <char **> &buf, &len_buf)
+    len_buf = len(py_buf)
+    if type(py_buf) == bytearray or type(py_buf) == str:
+        py_buf = bytes(py_buf)
```

Comment:
This doesn't look like it works as expected. I think this will raise `TypeError: string argument without an encoding` if you pass a string.

Reply:
Passing a string will raise that error in Py3. But I think Jeff did it to maintain Py2 compatibility because bytes are str's in Py2:

```
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from mtbl import *
>>> type(varint_encode(10000))
<type 'str'>
>>> type(b'\xde\xad\xbe\xef')
<type 'str'>
>>>
```

Reply:
I tried to come up with a way to fix the TypeError in Py3, but still have it work in Py2:

```
if type(py_buf) == bytearray:
    py_buf = bytes(py_buf)
elif type(py_buf) == str:
    try:
        py_buf = bytes(py_buf.encode()) #py3 and py2 (e.g. if py_buf = 'foo')
    except UnicodeDecodeError:
        pass #py2 where bytes are strings (e.g. type(b'\xc4\xb8\xd30\x00\x00\x00') == str)
```

What do you think?